testing/device_reference.cu

```diff
@@ -218,14 +219,37 @@ void TestDeviceReferenceSwap(void)
   ref2 = 13;
 
   // test thrust::swap()
-  thrust::swap(ref1, ref2);
+  swap(ref1, ref2);
```

Comment:
I think the point of this code is to test `thrust::swap`, no?

The existence of `thrust::swap` is a bit unfortunate. If you try to `swap` a type that has both `std::` and `thrust::` as associated namespace, the compiler has no way to pick between `std::swap` and `thrust::swap`. We could consider defining `thrust::swap` as a global function _object_ that dispatches to (unqualified) `swap` in a context that has brought `std::swap` in with a `using` declaration. Then a simple call to `thrust::swap` will do the ADL lookup internally, and the potential ambiguity between `std::swap` and `thrust::swap` goes away. I think that's a change that is unlikely to break anybody.

@allisonvacanti thoughts?

Reply:
Good catch, your proposed fix sounds good to me so long as nothing is specializing `thrust::swap`.

The `THRUST_INLINE_CONSTANT` macro may be needed when defining the function object, see https://github.com/NVIDIA/thrust/blob/main/thrust/detail/config/cpp_compatibility.h#L47-L74